File: src/DepositfilesSDK/DepositfilesSDK/utilitiez.vb
Imports System.ComponentModel

Public Class utilitiez

    Public Shared Function AsQueryString(parameters As Dictionary(Of String, String)) As String
        If Not parameters.Any() Then Return String.Empty
        Dim builder = New Text.StringBuilder("?")
        Dim separator = String.Empty
        For Each kvp In parameters.Where(Function(P) Not String.IsNullOrEmpty(P.Value))
            builder.AppendFormat("{0}{1}={2}", separator, Net.WebUtility.UrlEncode(kvp.Key), Net.WebUtility.UrlEncode(kvp.Value.ToString()))
            separator = "&"
        Next
        Return builder.ToString()
    End Function

    Shared Function ExtractLinks(HTMLString As String, HTTPS As Boolean) As List(Of String)
        Dim linkParser = New Text.RegularExpressions.Regex(String.Format("\b(?:{0}?://|www\.)\S+\b", If(HTTPS, "https", "http")), Text.RegularExpressions.RegexOptions.Compiled Or Text.RegularExpressions.RegexOptions.IgnoreCase)
        Dim linkz = linkParser.Matches(HTMLString)
        Return (From mch As Text.RegularExpressions.Match In linkz Select mch.Value).ToList
    End Function

    Enum UploadTypes
        FilePath
        Stream
        BytesArry
    End Enum
    Enum SearchTypeEnum
        Contains
        Exact
        StartWith
        EndWith
        Ext
    End Enum
End Class

Public Class ConnectionSettings
    Public Property TimeOut As System.TimeSpan = Nothing
    Public Property CloseConnection As Boolean? = True
    Public Property Proxy As ProxyConfig = Nothing
End Class

File: DepositfilesSDK/DepositfilesSDK/Utilitiez.cs
using System.Collections.Generic;
using System.Linq;

namespace DepositfilesSDK
{
    public class Utilitiez
    {

        public static string AsQueryString(Dictionary<string, string> parameters)
        {
            if (!parameters.Any())
            {
                return string.Empty;
            }

            var builder = new System.Text.StringBuilder("?");
            var separator = string.Empty;
            foreach (var kvp in parameters.Where(P => !string.IsNullOrEmpty(P.Value)))
            {
                builder.AppendFormat("{0}{1}={2}", separator, System.Net.WebUtility.UrlEncode(kvp.Key), System.Net.WebUtility.UrlEncode(kvp.Value.ToString()));
                separator = "&";
            }
            return builder.ToString();
        }

        public static List<string> ExtractLinks(string HTMLString, bool HTTPS)
        {
            var linkParser = new System.Text.RegularExpressions.Regex(string.Format(@"\b(?:{0}?://|www\.)\S+\b", HTTPS ? "https" : "http"), System.Text.RegularExpressions.RegexOptions.Compiled | System.Text.RegularExpressions.RegexOptions.IgnoreCase);
            System.Text.RegularExpressions.MatchCollection linkz = linkParser.Matches(HTMLString);
            List<string> tobereturn = new List<string>();
            return (from System.Text.RegularExpressions.Match mch in linkz select mch.Value).ToList();
        }

        public enum UploadTypes
        {
            FilePath,
            Stream,
            BytesArry
        }

        public enum SearchTypeEnum
        {
            Contains,
            Exact,
            StartWith,
            EndWith,
            Ext
        }

    }
}
